### PR TITLE
Make the app cross-platform by using Path.Join

### PR DESCRIPTION
**Pull request by amir734jj**
- Fixed login bug by using OR instead of AND (probably a logic bug)
- Used `await using` instead of `using` which is a new C# feature
- Added support for Linux and Mac by using cross-platform `Path.Join` as oppose to string interpolation

### CODE REVIEW

File: Deployr.Web/Logic/DeployLogic.cs [hunk 1]
```diff
@@ -58,39 +58,39 @@ public async Task<BasicResponse> UploadPackageAsync(int deploymentId, IFormFile
 
 		public async Task<BasicResponse> CreateDeployment(CreateDeploymentRequest metadata)
 		{
-			using var dataContext = await _dataContextFactory.Construct();
+			await using var dataContext = await _dataContextFactory.Construct();
```
Comment (TahaNKhan) [inline Deployr.Web/Logic/DeployLogic.cs:61]: Ooo fancy

File: Deployr.Web/Logic/DeployLogic.cs [hunk 2]
```diff
@@ -108,7 +108,7 @@ private DirectoryInfo ValidateAndCleanDirectory(string deployLocation, bool forc
 				throw new WebException(400, "Package already deployed");
 
 			// Delete old artifacts
-			if (directoryExists && forceDeploy)
+			if (directoryExists || forceDeploy)
```
Comment (TahaNKhan) [inline Deployr.Web/Logic/DeployLogic.cs]: We should make sure the directory exists before we delete it, right?

File: Deployr.Web/Logic/DeployLogic.cs [hunk 3]
```diff
@@ -117,20 +117,24 @@ private DirectoryInfo ValidateAndCleanDirectory(string deployLocation, bool forc
...
 		private static string GetDeployLocation(string packageName, string version)
 		{
-			return $"{DetermineDefaultDeployLocation()}\\{packageName}-{version}";
+			return Path.Join(DetermineDefaultDeployLocation(), $"{packageName}-{version}");
 		}
 
 		private static string DetermineDefaultDeployLocation()
 		{
 			const string windowsDefaultDeployLocation = "C:\\CustomApps";
+			const string linuxDefaultDeployLocation = "~/CustomApps";
```
Comment (TahaNKhan) [inline Deployr.Web/Logic/DeployLogic.cs]: Would /use/local/bin be appropriate here? I’m not too familiar with Linux directory structure anymore.

**Comment by amir734jj** [comment]
Updated the code. Removed `forceDeploy` because it's redundant in the second if statement. Rider will show it's reducent because it will be always true.

**Review by TahaNKhan — APPROVED** [review]

**Pull request merged**